src/lib/utils/blend.js

```diff
@@ -0,0 +1,23 @@
+// get current blending settings
+export function getBlendMode(gl) {
```

Comment:
>  I found this list of getParameter calls takes an large proportion of execution time

@shaojingli Yes `getParameter` requires a roundtrip from the GPU and slows things down, just like `getError` does in debug mode. I have a solution in a luma.gl PR, a WebGL state manager that allows the app to push and pop state without ever hitting `getParameter`. Perhaps we could work together on getting this landed soon?